src/Device/Driver/XCVario.cpp

```diff
@@ -31,33 +31,54 @@ Copyright_License {
 #include "NMEA/InputLine.hpp"
 #include "util/Clamp.hpp"
 #include "Atmosphere/Pressure.hpp"
+#include "Operation/Operation.hpp"
 #include <math.h>
 
+#define XCV_VERSION_UNKNOWN 0  // as long device has not yet answered and no timeout
+#define XCV_VERSION_1  1       // exchange legacy format with fractional ballast in $PXCV plus CAI302 !g command
+#define XCV_VERSION_2  2       // exchange absolute ballast in liters by !xc[s|v] messages
+
 class XVCDevice : public AbstractDevice {
   Port &port;
 
+private:
+  int protocol_version = XCV_VERSION_UNKNOWN;  // controls the
```

Comment:
Why is this a signed integer? Can it be negative?

Reply:
will be the corresponding enun type later.

Reply:
unsigned now here:
https://github.com/XCSoar/XCSoar/blob/2b7ae12843d31dfd27a118d97ee66919a375fbe1/src/Device/Driver/XCVario.cpp#L45
and initialized in constructor:
https://github.com/XCSoar/XCSoar/blob/2b7ae12843d31dfd27a118d97ee66919a375fbe1/src/Device/Driver/XCVario.cpp#L55